pkg/tls/cred_test.go

```diff
@@ -47,3 +50,68 @@ func TestCredEncodeCertificateAndTrustChain(t *testing.T) {
 		t.Errorf("Encoded Certificate And TrustChain does not match expected output")
 	}
 }
+
+func TestCrtExpiry(t *testing.T) {
+	root := newRoot(t)
+	rootTrust := root.Cred.Crt.CertPool()
+
+	cred, err := root.GenerateEndEntityCred("expired.test")
+	if err != nil {
+		t.Fatalf("failed to create end entity cred: %s", err)
+	}
+
+	crt, err := DecodePEMCrt(cred.Crt.EncodePEM())
+	if err != nil {
+		t.Fatalf("Failed to decode PEM Crt: %s", err)
+	}
+
+	i := x509.NewCertPool()
+	for _, c := range crt.TrustChain {
+		i.AddCert(c)
+	}
+	vo := x509.VerifyOptions{Roots: rootTrust, Intermediates: i, DNSName: "expired.test"}
+
+	testCases := []struct {
+		notBefore    time.Time
+		notAfter     time.Time
+		shouldExpire bool
+	}{
+		//cert not valid yet
+		{
+			notAfter:     time.Now().AddDate(0, 0, 20),
+			notBefore:    time.Now().AddDate(0, 0, 10),
+			shouldExpire: true,
```

Comment:
I think the tests should be a bit more specific. Namely. The firrst two tests only check whether the error is of type `expired`. However they dont check the specific type of time constraints violation. Is it really not valid yet? I think now that you control the type of error message emitted, you can simply go ahead and test on that. You know what the error format will be so in your test case struct instead of simply having a boolean to expect upon, have an err type or a string and check the explicit error message. No need to use the `crtExpiryError`